src/form.js

```diff
@@ -802,7 +802,32 @@ define([
             }
         },
         changeMugType: function (mug, questionType) {
-            this.mugTypes.changeType(mug, questionType);
+            var _this = this;
+            _this.mugTypes.changeType(mug, questionType);
+            _this.setMugActions(mug);
+        },
+        setMugActions: function(mug) {
+            var _this = this,
+                tree = _this.vellum.data.core.$tree,
+                action_id = "add_choice";
+
+            if (mug.__className === 'Select' || mug.__className === 'MSelect') {
```

Comment:
Rather than hard-coding mug type names here, what do you think of adding attributes to the mug type options (like `mug.options.canAddChoices`)?